core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -48,8 +48,10 @@
   public static final String ICEBERG_CATALOG_TYPE = "type";
   public static final String ICEBERG_CATALOG_TYPE_HADOOP = "hadoop";
   public static final String ICEBERG_CATALOG_TYPE_HIVE = "hive";
+  public static final String ICEBERG_CATALOG_TYPE_JDBC = "jdbc";
```

Comment:
I don't think we need all the logic in this file, it's much cleaner for everything other than hive and hadoop to be loaded through impl. It makes the user experience much more consistent, and we don't need to keep updating the list of types in documentation.

Reply:
makes sense, rolled the change back